CHANGELOG.md

```diff
@@ -1,168 +1,249 @@
-`DeviceSegmentedRadixSort` and `DeviceSegmentedSort`.
-- CUB backend references CUB and Thrust version 1.17.1.
-
-### Known Issues
-- `BlockRadixRankMatch` is currently broken under the rocPRIM backend.
-- `BlockRadixRankMatch` with a warp size that does not exactly divide the block size is broken under the CUB backend.
+
+### Additions
+
+* CMake functionality improves build parallelism for the test suite that splits compilation units by
+function or parameters
+* New overload for `BlockAdjacentDifference::SubtractLeftPartialTile`, which takes a predecessor item
+
+### Changes
+
+* Improved build parallelism of the test suite by splitting up large compilation units for `DeviceRadixSort`, `DeviceSegmentedRadixSort`, and `DeviceSegmentedSort`
+* The CUB backend references CUB and Thrust version 1.17.1
+
+### Known issues
+
+* `BlockRadixRankMatch` is currently broken for the rocPRIM backend
+* `BlockRadixRankMatch` with a warp size that does not divide exactly by the block size is broken for
+  the CUB backend
 
 ## hipCUB-2.12.0 for ROCm 5.3.0
-### Added
-- UniqueByKey device algorithm
-- SubtractLeft, SubtractLeftPartialTile, SubtractRight, SubtractRightPartialTile overloads in BlockAdjacentDifference.
-  - The old overloads (FlagHeads, FlagTails, FlagHeadsAndTails) are deprecated.
-- DeviceAdjacentDifference algorithm.
-- Extended benchmark suite of `DeviceHistogram`, `DeviceScan`, `DevicePartition`, `DeviceReduce`,
-`DeviceSegmentedReduce`, `DeviceSegmentedRadixSort`, `DeviceRadixSort`, `DeviceSpmv`, `DeviceMergeSort`,
-`DeviceSegmentedSort`
-### Changed
-- Obsolated type traits defined in util_type.hpp. Use the standard library equivalents instead.
-- CUB backend references CUB and thrust version 1.16.0.
-- DeviceRadixSort's num_items parameter's type is now templated instead of being an int.
-  - If an integral type with a size at most 4 bytes is passed (i.e. an int), the former logic applies.
-  - Otherwise the algorithm uses a larger indexing type that makes it possible to sort input data over 2**32 elements.
+
+### Additions
+
+* `UniqueByKey` device algorithm
+* `SubtractLeft`, `SubtractLeftPartialTile`, `SubtractRight`, and `SubtractRightPartialTile` overload in
+  `BlockAdjacentDifference`
+  * The old overloads (`FlagHeads`, `FlagTails`, `FlagHeadsAndTails`) are deprecated
+* `DeviceAdjacentDifference` algorithm
+* Extended benchmark suite of `DeviceHistogram`, `DeviceScan`, `DevicePartition`, `DeviceReduce`,
+`DeviceSegmentedReduce`, `DeviceSegmentedRadixSort`, `DeviceRadixSort`, `DeviceSpmv`,
+`DeviceMergeSort`, and `DeviceSegmentedSort`
+
+### Changes
+
+* Obsolete type traits defined in `util_type.hpp`; use the standard library equivalents instead
+* CUB backend references CUB and Thrust version 1.16.0
+* `DeviceRadixSort` `num_items` parameter type is now templated instead of being an int
+  * If an integral type with a maximum size of 4 bytes is passed (an int), the former logic applies;
```

Comment:
Does this bullet point look correct? (e.g., "maximum size of 4 bytes" and "2**32 elements")